tests/macsec/test_double_switch_reboot.py

```diff
@@ -0,0 +1,94 @@
+import pytest
+import logging
+import time
+
+from tests.common.reboot import reboot
+from tests.common.utilities import wait_until
+from tests.common.platform.interface_utils import \
+    check_interface_status_of_up_ports
+from tests.common.platform.processes_utils import wait_critical_processes
+from tests.macsec.macsec_helper import check_appl_db
+
+SONIC_SSH_PORT = 22
+SONIC_SSH_REGEX = 'OpenSSH_[\\w\\.]+ Debian'
+
+logger = logging.getLogger(__name__)
+post_reboot_time = 240
+
+pytestmark = [
+    pytest.mark.macsec_required,
+    pytest.mark.topology("any")
+]
+
+
+def compare_bgp_routes(dut, v4_rt_cnt_1, v6_rt_cnt_1, v4_rt_cnt_2, v6_rt_cnt_2):
+    logger.info("{}".format(dut))
+    logger.info("IPv4 routes: pre-reboot {} post-reboot {}".format(
+        v4_rt_cnt_1, v4_rt_cnt_2))
+    logger.info("IPv6 routes: pre-reboot {} post-reboot {}".format(
+        v6_rt_cnt_1, v6_rt_cnt_2))
+    return v4_rt_cnt_1 == v4_rt_cnt_2 and v6_rt_cnt_1 == v6_rt_cnt_2
+
+
+def get_bgp_routes(duthost):
+    sumv4, sumv6 = duthost.get_ip_route_summary(skip_kernel_tunnel=True)
+    totalsv4 = sumv4.get('Totals', {})
+    totalsv6 = sumv6.get('Totals', {})
+    routesv4 = totalsv4.get('routes', 0)
+    routesv6 = totalsv6.get('routes', 0)
+    return routesv4, routesv6
+
+
+def reboot_nbr(nbrhost):
+    # Save the original config file
+    nbrhost.shell("cp /etc/sonic/config_db.json config_db.json")
+    # Save the current config file
+    nbrhost.shell("sonic-cfggen -d --print-data > /etc/sonic/config_db.json")
```

Comment:
No need to save config since the test doesn't change any config in this test case. Even if you want to save config, you should use "sudo config save -y" for simplicity.